### PR TITLE
MCOL-5180 Check CPU vector instructions set in installer and PrimProc on start

### PR DESCRIPTION
**Pull request by mariadb-AndreyPiskunov**
Adding various checks for simd architecture support.

### CODE REVIEW

File: primitives/primproc/archcheck.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/* Copyright (C) 2014 InfiniDB, Inc.
```
Comment (mariadb-LeonidFedorov) [inline primitives/primproc/archcheck.h]: it's 2022 now, and Copyright (C) 2022 MariaDB Corporation

Comment (mariadb-AndreyPiskunov) [reply primitives/primproc/archcheck.h]: Sorry, my bad. Just copied it from another file and didn't look.

File: primitives/primproc/archcheck.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+__attribute__ ((target ("sse4.2")))
+int checkArchitecture ()
+{
+  // version for SSE4.2
+  return 2;
+}
+#elif defined(__aarch64__)
+int checkArchitecture()
```
Comment (mariadb-LeonidFedorov) [inline primitives/primproc/archcheck.h]: let's replace int with enum for better readability

File: primitives/primproc/primproc.cpp [hunk 3]
```diff
@@ -735,6 +738,11 @@ int ServicePrimProc::Child()
 
 int main(int argc, char** argv)
 {
+  if (checkArchitecture() != arcitecture::SSE4_2 || checkArchitecture() != arcitecture::ASIMD)
```
Comment (mariadb-LeonidFedorov) [inline primitives/primproc/primproc.cpp]: This is always true expression

File: primitives/primproc/archcheck.h (hunk 2 above)
Comment (mariadb-AndreyPiskunov) [reply primitives/primproc/archcheck.h]: Done

File: primitives/primproc/primproc.cpp (hunk 3 above)
Comment (mariadb-LeonidFedorov) [reply primitives/primproc/primproc.cpp]: Couldn't resist to make a commit

File: debian/mariadb-plugin-columnstore.preinst [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -e
+
+# Check for sse4.2 or neon support
+
+if [ -z $(cat /proc/cpuinfo | grep -E 'sse4_2|neon|asimd' | head -c1) ]; then
+    echo "No support found for sse4 or neon simd instructions; aborting"
```
Comment (mariadb-LeonidFedorov) [inline debian/mariadb-plugin-columnstore.preinst]: Please write more stupid-proof and polite error message in both deb and rpm installators
like
'error: this machine CPU lacks of support for Intel SSE4.2 or ARM Advanced SIMD instructions. Columnstore requires one of this instruction sets, installation aborted

**Comment by mariadb-LeonidFedorov** [comment]
LGTM. Waiting for docker tests, and something happen with regression tests. I started clean develop build to compare

**Review by mariadb-LeonidFedorov — APPROVED** [review]

**Pull request merged**